src/query-builder/transformer/DocumentToEntityTransformer.ts

```diff
@@ -13,10 +15,12 @@ export class DocumentToEntityTransformer {
     // -------------------------------------------------------------------------
 
     constructor(// private selectionMap: AliasMap,
-                // private joinMappings: JoinMapping[],
-                // private relationCountMetas: RelationCountAttribute[],
-                private enableRelationIdValues: boolean = false
+        // private joinMappings: JoinMapping[],
+        // private relationCountMetas: RelationCountAttribute[],
+        private queryRunner: MongoQueryRunner,
```

Comment:
I really do not like pass `QueryRunner` into `DocumentToEntityTransformer` class. This class have only one simple purpose - convert object to entity - so we should keep it simple.

Reply:
But `QueryRunner` pass into other transformers too

Reply:
Sure in `RawSqlResultsToEntityTransformer` but it is used to sync create entity not to async operations like broadcasting. This is not the root problem.